iree/turbine/kernel/boo/ops/utils.py

```diff
@@ -115,21 +116,81 @@ def get_func_name(
     return "_".join(name_items)
 
 
-def _tensor_type_str(t: torch.Tensor | None) -> str:
-    if t is None:
-        return ""
-    shape = t.shape
-    dtype = str(t.dtype).removeprefix("torch.")
+def _tensor_type_str(shape: Iterable[int], dtype: torch.dtype) -> str:
+    dtype = str(dtype).removeprefix("torch.")
     shape_str = "x".join([str(dim) for dim in shape])
     return shape_str + f"x{dtype}"
 
 
+def _is_contiguous(
+    t: torch.Tensor | TensorMetadata, memory_format: torch.memory_format
+) -> bool:
+    if isinstance(t, torch.Tensor):
+        return t.is_contiguous(memory_format=memory_format)
+    if isinstance(t, TensorMetadata):
+        return t.memory_format == memory_format
+    raise TypeError(
+        f"Unhandled type: {type(t)}. _is_contiguous input 0 must be a torch.Tensor or TensorMetadata object."
+    )
```

Comment:
since `t` is typed as `torch.Tensor | TensorMetadata`, you don't really need to check the type here (the type checker should infer `t` is `TensorMetadata` here and be happy)
```suggestion
    return t.memory_format == memory_format
```

Reply:
I'm fine with that, but the typing isn't a guarantee. In fact, mypy isn't even running in `iree/turbine/kernel` right now. 

This check won't happen at runtime, so I don't see any harm in keeping it.